zenodo/modules/utils/cli.py

```diff
@@ -440,3 +445,45 @@ def retry_indexing_failed_openaire_records(eager):
         else:
             openaire_direct_index.delay(
                 str(recid.object_uuid), retry=False)
+
+
+@utils.command('delete_openaire_records')
+@with_appcontext
+def delete_openaire_records():
+    """Deletes half deleted OpenAIRE records."""
+    for key in _iter_openaire_direct_index_keys():
+        try:
+            RECID = key.split("openaire_direct_index:")[1]
+
+            record_data = PersistentIdentifier.query.filter_by(
+                pid_type="recid", pid_value=str(RECID), status=PIDStatus.DELETED
+            ).one()
+            record = ZenodoRecord.get_record(
+                    record_data.object_uuid, with_deleted=True
+                    )
+            if "removal_reason" in record:
+                last_valid_revision = record.revisions[-2]
```

Comment:
There are some rare cases where the -2th (i.e. previous revision) won't do it, because of past mistakes. A more "natural" way to find the last good revision, is to seek backwards, until e.g. you find a revision that has a `resource_type` field (which is actually what we're looking for to figure out the OpenAIRE IDs).